synapse/lib/stormlib/cortex.py

```diff
@@ -0,0 +1,656 @@
+        $lib.print(`Set HTTP API {$api.iden} to index {$index}`)
+        '''
+    }
+]
+
+class HttpApi(s_stormtypes.StormType):
+    '''
+    Extended HTTPApi object
+    '''
+    _storm_typename = 'http:api'
+    _storm_locals = ()
+
+    # TODO DOCSTRING
+    # TODO LOCALS
+
+    def __init__(self, runt, info):
+        s_stormtypes.StormType.__init__(self)
+        self.runt = runt
+        self.info = info
+        self.iden = self.info.get('iden')
+
+        self.stors.update({
+            # General helpers
+            'name': self._storName,
+            'desc': self._storDesc,
+            'path': self._storPath,
+            'vars': self._storVars,
+            'view': self._storView,
+            'runas': self._storRunas,
+            'owner': self._storOwner,
+            'perms': self._storPerms,
+            'readonly': self._storReadonly,
+            'authenticated': self._storAuthenticated,
+        })
+
+        self.gtors.update({
+            'name': self._gtorName,
+            'desc': self._gtorDesc,
+            'path': self._gtorPath,
+            'view': self._gtorView,
+            'runas': self._gtorRunas,
+            'owner': self._gtorOwner,
+            'perms': self._gtorPerms,
+            'readonly': self._gtorReadonly,
+            'authenticated': self._gtorAuthenticated,
+        })
+
+        self.ctors.update({
+            'vars': self._ctorVars,
+            'methods': self._ctorMethods
+        })
+
+        # constants
+        self.locls.update(self.getObjLocals())
+        self.locls.update({
+            'iden': self.iden,
+        })
+
+    def getObjLocals(self):
+        return {
+            'pack': self._methPack,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _methPack(self):
+        return copy.deepcopy(self.info)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    def _ctorMethods(self, path):
+        return HttpApiMethods(self)
+
+    async def _storPath(self, path):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        path = await s_stormtypes.tostr(path)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'path', path)
+        self.info['path'] = path
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorPath(self):
+        return self.info.get('path')
+
+    async def _storName(self, name):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        name = await s_stormtypes.tostr(name)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'name', name)
+        self.info['name'] = name
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorView(self):
+        iden = self.info.get('view')
+        vdef = await self.runt.snap.core.getViewDef(iden)
+        if vdef is None:
+            raise s_exc.NoSuchView(mesg=f'No view with {iden=}', iden=iden)
+        return s_stormtypes.View(self.runt, vdef, path=self.path)
```

Comment:
I leaned into returning a heavy object here since the first thing someone may do with an iden is to get the heavy view object; and if that doesn't exist in `$lib.view.get()` it will throw a NoSuchView exception.

May want to return the iden and let the caller handle it?

Reply:
:+1: good call. heavy object is best.